### PR TITLE
Justin Pasko

### PR DESCRIPTION
**Pull request by jpasko1112**

### CODE REVIEW

File: lib/vehicle_factory.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    details = {
+                vin: vehicle[:vin_1_10],
+                year: vehicle[:model_year],
+                make: vehicle[:make],
+                model: vehicle[:model],
+                engine: :ev
+      }
+      Vehicle.new(details)
+    end
+  end
```
Comment (megstang) [inline lib/vehicle_factory.rb:15]: THis method looks good! NIce work

File: lib/facility.rb [hunk 2]
```diff
@@ -1,14 +1,64 @@
...
+    else false
+    end
+  end
+
+  def add_collected_fees(vehicle)
+    if vehicle.antique?
+      @collected_fees += 25
+    elsif vehicle.electric_vehicle?
+      @collected_fees += 200
+    else @collected_fees += 100
```
Comment (megstang) [inline lib/facility.rb:60]: put `@collected_fees += 100` on the line below this `else`

File: lib/facility.rb [hunk 3]
```diff
@@ -1,14 +1,64 @@
...
+  def add_service(service)
     @services << service
   end
+
+  def register_vehicle(vehicle)
+    if @services.include?('Vehicle Registration')
+    @registered_vehicles << vehicle
+    vehicle.registered_date
+    vehicle.assign_plate
+    add_collected_fees(vehicle)
```
Comment (megstang) [inline lib/facility.rb:28]: nice, I like this helper method you created!

File: lib/dmv_facility.rb [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+                })
+    end
+  end
+
+  #not sure if this actually returns the address the way it should. I couldn't figure out how to print it or view it in pry.
+  def create_facility_mo(offices)
+    offices.map do |office|
+      Facility.new ({
+                  name: office[:name],
+                  address: office[:address1, :city, :state, :zip],
```
Comment (megstang) [inline lib/dmv_facility.rb:28]: we would need to do some sort of string manipulation to get the address formatted. For example, `office[:address1] + " " + office[:city] + " " ......`

**Review by megstang — COMMENTED** [review]
@jpasko1112 I left some comments on your code. Please let me know if you have any questions. I'll follow up in slack with more feedback later today.

File: lib/facility.rb (hunk 2 above)
Comment (jpasko1112) [reply lib/facility.rb:60]: else
  @collected_fees += 100(?)

File: lib/facility.rb (hunk 3 above)
Comment (jpasko1112) [reply lib/facility.rb:28]: Does this negate what you said earlier about including the assign date to the  register method?

File: lib/dmv_facility.rb (hunk 4 above)
Comment (jpasko1112) [reply lib/dmv_facility.rb:28]: Got it, thank you!

File: lib/registrant.rb [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    if permit?
+      @license_data[:written] = true
+    end
+  end
+
+  def license?
+    if written?
+      @license_data[:license] = true
+    end
+  end
```
[the remark this answers is not in the document]
Comment (jpasko1112) [reply lib/registrant.rb:36]: Ok, thank you.

**Pull request closed without merging**